snapcraft/internal/meta/_snap_packaging.py

```diff
@@ -402,6 +402,9 @@ def _compose_snap_yaml(self):
         snap_yaml['version'] = _version.get_version(
             self._config_data['version'],
             self._config_data.get('version-script'))
+        if not isinstance(snap_yaml['version'], str):
+            logger.warn('"version" property is not a string: {!r}'.format(
```

Comment:
Why do I as a random user care if it's a string or not?

Reply:
I changed my approach to parsing float as string instead of a warning.